code/modules/modular_computers/file_system/programs/radar.dm

```diff
@@ -0,0 +1,291 @@
+/datum/computer_file/program/radar/process()
+	if(computer.active_program != src)
+		STOP_PROCESSING(SSfastprocess, src) //We're not the active program, it's time to stop.
+		return
+	if(!selected)
+		return
+
+	var/atom/movable/signal = find_atom()
+	if(!trackable(signal))
+		program_icon_state = "[initial(program_icon_state)]lost"
+		if(last_icon_state != program_icon_state)
+			computer.update_icon()
+			last_icon_state = program_icon_state
+		return
+
+	var/here_turf = get_turf(computer)
+	var/target_turf = get_turf(signal)
+	var/trackdistance = get_dist_euclidian(here_turf, target_turf)
+	switch(trackdistance)
+		if(0)
+			program_icon_state = "[initial(program_icon_state)]direct"
+		if(1 to 12)
+			program_icon_state = "[initial(program_icon_state)]close"
+		if(13 to 24)
+			program_icon_state = "[initial(program_icon_state)]medium"
+		if(25 to INFINITY)
+			program_icon_state = "[initial(program_icon_state)]far"
+
+	if(last_icon_state != program_icon_state)
+		computer.update_icon()
+		last_icon_state = program_icon_state
+	computer.setDir(get_dir(here_turf, target_turf))
+
+//We can use process_tick to restart fast processing, since the computer will be running this constantly either way.
+/datum/computer_file/program/radar/process_tick()
+	if(computer.active_program == src)
+		START_PROCESSING(SSfastprocess, src)
+
+///////////////////
+//Suit Sensor App//
+///////////////////
+
+///A program that tracks crew members via suit sensors
+/datum/computer_file/program/radar/lifeline
+	filename = "lifeline"
+	filedesc = "Lifeline"
+	extended_desc = "This program allows for tracking of crew members via their suit sensors."
+	requires_ntnet = TRUE
+	transfer_access = ACCESS_MEDICAL
+	available_on_ntnet = TRUE
+
+/datum/computer_file/program/radar/lifeline/find_atom()
+	return locate(selected) in GLOB.carbon_list //currently we dont have a list of humanoids so this'll have to do
+
+/datum/computer_file/program/radar/lifeline/scan()
+	if(world.time < next_scan)
+		return
+	next_scan = world.time + (2 SECONDS)
+	objects = list()
+	for(var/i in GLOB.carbon_list)
+		var/mob/living/carbon/human/humanoid = i
+		if(!trackable(humanoid))
+			continue
+		var/crewmember_name = "Unknown"
+		if(humanoid.wear_id)
+			var/obj/item/card/id/ID = humanoid.wear_id.GetID()
+			if(ID?.registered_name)
+				crewmember_name = ID.registered_name
+		var/list/crewinfo = list(
+			ref = REF(humanoid),
+			name = crewmember_name,
+			)
+		objects += list(crewinfo)
+
+/datum/computer_file/program/radar/lifeline/trackable(mob/living/carbon/human/humanoid)
+	if(!humanoid || !istype(humanoid))
+		return FALSE
+	if(..())
+		if(humanoid in SSnanites.nanite_monitored_mobs)
+			return TRUE
+		if(istype(humanoid.w_uniform, /obj/item/clothing/under))
+			var/obj/item/clothing/under/uniform = humanoid.w_uniform
+			if(uniform.has_sensor && uniform.sensor_mode >= SENSOR_COORDS) // Suit sensors must be on maximum
+				return TRUE
+	return FALSE
```

Comment:
This needs to be the same as the crew monitor and check things like is_jammed()

Reply:
signal jammers suck